<?xml version="1.0" ?>

<robot name="mrm" xmlns:xacro="http://www.ros.org/wiki/xacro">
    <link name="base_link">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0.1"/>
            <geometry>
                <box size="0.25 0.25 0.2"/>
            </geometry>
        </visual>
    </link>    
    <joint name="joint1" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5" />
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <parent link="base_link"/>
        <child link="link1"/>
    </joint>    
    <link name="link1">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0.225"/>
            <geometry>
                <cylinder radius="0.1" length="0.05"/>
            </geometry>
        </visual>
    </link>
    <joint name="joint2" type="revolute">
        <axis xyz="0 1 0" />
        <limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5" />
        <origin rpy="0 0 0" xyz="0 0 0.275"/>
        <parent link="link1"/>
        <child link="link2"/>
    </joint>    
    <link name="link2">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0.25"/>
            <geometry>
                <cylinder radius="0.035" length="0.45"/>
            </geometry>
        </visual>
    </link>
    <joint name="joint3" type="revolute">
        <axis xyz="0 1 0" />
        <limit effort="1000.0" lower="-1.57" upper="1.57" velocity="0.5" />
        <origin rpy="0 0 0" xyz="0 0 0.475"/>
        <parent link="link2"/>
        <child link="link3"/>
    </joint>    
    <link name="link3">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0.25"/>
            <geometry>
                <cylinder radius="0.035" length="0.45"/>
            </geometry>
        </visual>
    </link>
    <joint name="joint4" type="revolute">
        <axis xyz="0 1 0" />
        <limit effort="1000.0" lower="0" upper="1.57" velocity="0.5" />
        <origin rpy="0 0 0" xyz="0 0 0.475"/>
        <parent link="link3"/>
        <child link="link4"/>
    </joint>    
    <link name="link4">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0.175"/>
            <geometry>
                <cylinder radius="0.035" length="0.275"/>
            </geometry>
        </visual>
    </link>
    <joint name="end" type="fixed">        
        <origin rpy="0 0 0" xyz="0 0 0.4"/>
        <parent link="link4"/>
        <child link="final_link"/>
    </joint>    
    <link name="final_link">
        <visual>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <geometry>
                <sphere radius="0.035"/>
            </geometry>
        </visual>
    </link>

</robot>



